tests/projects/test_project.py

```diff
@@ -443,6 +443,81 @@ def test_set_func_requirements():
     ]
 
 
+def test_set_function_mask_v3io_credentials():
+    project = mlrun.projects.MlrunProject("newproj")
+    describe_func1 = mlrun.import_function("hub://describe", project=project.name)
+    v3io_access_key = "v3io-access-key"
+    v3io_username = "v3io-username"
+
+    os.environ["V3IO_ACCESS_KEY"] = v3io_access_key
+    os.environ["V3IO_USERNAME"] = v3io_username
+
+    describe_func1.apply(mlrun.platforms.auto_mount())
+    assert describe_func1.get_env("V3IO_ACCESS_KEY") is not None
+    assert describe_func1.get_env("V3IO_USERNAME") is not None
+    assert describe_func1.spec.volumes[0]["name"] == "v3io"
+    assert (
+        describe_func1.spec.volumes[0]["flexVolume"]["options"]["accessKey"]
+        == v3io_access_key
+    )
+    assert describe_func1.spec.volume_mounts[0]["name"] == "v3io"
+    describe_func1.metadata.credentials.access_key = "some-access-key"
+
+    project.set_function(describe_func1, "describe")
+
+    masked_describe_func = project.get_function("describe")
+    assert masked_describe_func.metadata.credentials.access_key is None
+    assert masked_describe_func.get_env("V3IO_ACCESS_KEY") is None
+    assert masked_describe_func.get_env("V3IO_USERNAME") is None
+    assert masked_describe_func.spec.volumes == []
+    assert masked_describe_func.spec.volume_mounts == []
+
+    secret_name = "v3io-secret"
+    describe_func1 = mlrun.import_function(
+        "hub://describe", project=project.name, new_name="describe1"
+    )
+    describe_func1.apply(mlrun.platforms.mount_v3io(secret=secret_name))
```

Comment:
While you're at it - try to set the `name` parameter, and see what happens.

Reply:
changed to your suggestion and it works